stack-packs/packs/amp.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+/* eslint-disable max-len */
+
+'use strict';
+
+const i18n = require('../../lighthouse-core/lib/i18n/i18n.js');
+
+const ampIcon = `data:image/svg+xml,%3Csvg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256"%3E%3Cpath d="M171.887 116.28l-53.696 89.36h-9.728l9.617-58.227-30.2.047a4.852 4.852 0 01-4.855-4.855c0-1.152 1.07-3.102 1.07-3.102l53.52-89.254 9.9.043-9.86 58.317 30.413-.043a4.852 4.852 0 014.855 4.855c0 1.088-.427 2.044-1.033 2.854l.004.004zM128 0C57.306 0 0 57.3 0 128s57.306 128 128 128 128-57.306 128-128S198.7 0 128 0z" fill="%230379c4" fill-rule="evenodd"/%3E%3C/svg%3E`;
+
+const UIStrings = {
+  /** Additional description of a Lighthouse audit that tells the user how they can improve image loading by using webp in the context of AMP. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
+  uses_webp_images: 'Consider displaying all your `amp-img` components in WebP formats while specifying an appropriate fallback for other browsers. [Learn more](https://amp.dev/documentation/components/amp-img/#example:-specifying-a-fallback-image).',
+  /** Additional description of a Lighthouse audit that tells the user how images are automatically lazy loaded for the `AMP framewor`k. This is displayed after a user expands the section to see more. No character length limits. Links in (parenthesis) become link texts to additional documentation. */
+  offscreen_images: '`amp-img` automatically lazy-loads images outside the first viewport. There\'s nothing you need to do.',
```

Comment:
I don't really love the phrase "there's nothing you need to do". Should this be included if there is nothing to do? This audit should never fail then, right?
* If it does fail, what can someone do?
* If it doesn't ever fail, remove it?

Reply:
Yeah that's a good point. It technically can fail if the user uses a regular <img> tag that generates an invalid AMP page.

Changed it to reflect that 👍